app/models/ems_event.rb

```diff
@@ -58,14 +58,46 @@ def self.group_name(group)
     group[:name]
   end
 
-  def self.add_queue(meth, ems_id, event)
-    MiqQueue.submit_job(
-      :service     => "event",
-      :target_id   => ems_id,
-      :class_name  => "EmsEvent",
-      :method_name => meth,
-      :args        => [event],
-    )
+  def self.artemis_client
+    require "manageiq-messaging"
+    # FIXME: logger should be set at a global level
+    ManageIQ::Messaging.logger = _log
+    @artemis_client ||= begin
+      queue_settings = Settings.workers.worker_base.queue_worker_base.event_handler
+      connect_opts = {
+        :host       => queue_settings.queue_hostname,
+        :port       => queue_settings.queue_port.to_i,
+        :username   => queue_settings.queue_username,
+        :password   => queue_settings.queue_password,
+        :client_ref => "event_handler",
+      }
+
+      # caching the client works, even if the connection becomes unavailable
+      # internally the client will track the state of the connection and re-open it,
+      # once it's available again - at least thats true for a stomp connection
+      ManageIQ::Messaging::Client.open(connect_opts)
+    end
+  end
+
+  def self.add_queue(_meth, ems_id, event)
+    # FIXME: I dont like, we query a deep settings path on every call
+    #  but we want to be able to switch implementations, when config changes via the UI
```

Comment:
Prefer Settings.prototype.queue_type .  Since you want a top-level "switch" for now.

The deep call is not expensive, so I wouldn't worry about it.